File: Modeling .NET/Delphi/Coroutines/MyProc.pas
unit MyProc;

interface
uses Simulation;

type
  TMyProc = class(Coroutine)
  protected
    { Private Declarations }
    procedure Run; override;
  public
    Name : string;
    Next : TMyProc;
    constructor Create(AName : string);
  end;

implementation

constructor TMyProc.Create(AName : string);
begin
  inherited Create;
  // TODO: Add any constructor code here
  Name := AName;
end;

procedure TMyProc.Run;
var
  i : Integer;
begin
  inherited;
  for i := 0 to 4 do
  begin
    Console.WriteLine('Сопрограмма {0}: {1}', [Name, i]);
    Next.SwitchTo;
  end;
end;

end.

File: Modeling .NET/VB/Coroutines/Coroutines/MyProc.vb
Imports Simulation

Public Class MyProc
    Inherits Coroutine
    Public Name As String
    Public NNext As MyProc

    Public Sub New(aName as String)
        Name = aName
    End Sub

    Protected Overrides Sub Run()
        MyBase.Run()
        Dim i As Integer
        For i = 0 To 4
            Console.WriteLine("Сопрограмма {0}: {1}", Name, i)
            SwitchTo(NNext)
        Next
    End Sub
End Class
